airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -458,10 +459,16 @@ def _datetime_to_label_safe_datestring(datetime_obj):
 
     def _labels_to_key(self, labels):
         try:
+            try_num = 1
+            try:
+                try_num = int(labels.get('try_number', '1'))
+            except ValueError:
+                self.log.warn("could not get try_number as an int: %s", labels.get('try_number', '1'))
             return (
                 labels['dag_id'], labels['task_id'],
                 self._label_safe_datestring_to_datetime(labels['execution_date']),
-                labels['try_number'])
+                try_num
```

Comment:
So, if you upgrade a live system any pods without the label will return `try_num` 1. That's the crux of the fix (along with adding the try num label one new pods made), right?

Would be a spot for a pep8 recommended (plays nice with vcs) redundant comma
```suggestion
                try_num,
```

Reply:
Yes, older pods in a live system will not have this try number set, and needs to be defaulted to something reasonable.